Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -99,27 +97,57 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath)
                         $publicConfigStorageAccountName = $publicConfig.PublicConfig.StorageAccount
                         Write-Verbose "Found PublicConfig.StorageAccount= '$publicConfigStorageAccountName'"
 
-                        $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName
-                        if ($publicConfigStorageKey)
+                        if ($storageAccountKeysMap.containsKey($role))
                         {
-                            Write-Verbose "##[command]New-AzureStorageContext -StorageAccountName $publicConfigStorageAccountName -StorageAccountKey <key>"
-                            $storageContext = New-AzureStorageContext -StorageAccountName $publicConfigStorageAccountName -StorageAccountKey $publicConfigStorageKey.Primary
+                            Write-Verbose "##Getting diagnostics storage account name and key from passed as storage keys."
+
+                            Write-Verbose "##$storageAccountName = $publicConfigStorageAccountName"
+                            $storageAccountName = $publicConfigStorageAccountName
+                            
+                            Write-Verbose "##$storageAccountKey = $storageAccountKeysMap.Get_Item($role)"
```

Comment:
Do not write storage key in logs. If you want to set as variable, make it secure variable.